recipes-openxt/openxt-measuredlaunch/openxt-measuredlaunch/seal-system

```diff
@@ -46,36 +52,59 @@ Usage: seal-system [-h] [-s|-u|-f] [-ctpgbr path] -[48 object[,objects]]"
 EOF
 }
 
+# Usage: pcr_in_selection PCR-ID
+# Returns 0 if pcr-id PCR is in ${pcr_selection}, 1 otherwise.
+# ${pcr_selection} is defined in the 'Configurables' section, it contains the
+# list of PCRs defined in ${config_pcrs}.
 pcr_in_selection() {
-    pcr="$1"
+    local pcr="$1"
 
-    for p in $pcr_selection ; do
-        if [ $p -eq $pcr ] ; then
+    for p in ${pcr_selection}; do
+        if [ "${p}" -eq "${pcr}" ]; then
             return 0
         fi
     done
 
     return 1
 }
 
+# Usage: read_pcr_selection
+# Sanitize and print the content of file ${config_pcrs} on stdout.
+# ${config_pcrs} is defined in the 'Configurables' section.
 read_pcr_selection() {
     # strip out -r/-p to leave just the digits
-    sed -e 's/-[rp]//g' $config_pcrs
+    sed -e 's/-[rp]//g' "${config_pcrs}"
 }
 
-# Globals
-TPM_DEV="$(find /sys/class -name tpm0)/device"
+# Usage rotate_sealed_files
+# Rename the existing sealed key files to *.old.
+rotate_sealed_keyfiles() {
+    if [ -e "${sealed_key}" ]; then
```

Comment:
How about a loop like `for extension in "" .sha1 .pub.sha1 .sha256 .pub.sha256; do [...]`?